### PR TITLE
[CRM][DASH] Extend CrmOrch to support DASH resources.

### PR DESCRIPTION
**Pull request by oleksandrivantsiv**
**What I did**
Extend CrmOrch to support DASH resources.

**Why I did it**
Add a possibility to monitor and report DASH critical resources.

**How I verified it**
Run VS tests.
A new `tests/test_crm_dash.py` is added to cover DASH CRM functionality.
**Details if related**

### CODE REVIEW

**Comment by liat-grozovik** [comment]
@prsunny @theasianpianist  kindly reminder to review the PR?

File: orchagent/dash/dashaclorch.cpp [hunk 1]
```diff
@@ -703,7 +721,12 @@ task_process_status DashAclOrch::taskRemoveDashAclRule(
         return task_failed;
     }
     m_dash_acl_rule_table.erase(itr);
-    m_dash_acl_group_table[group_id].m_rule_count--;
+    --acl_group.m_rule_count;
+
+    CrmResourceType crm_resource = acl_group.m_ip_version == SAI_IP_ADDR_FAMILY_IPV4 ?
```
Comment (prsunny) [inline orchagent/dash/dashaclorch.cpp:726]: suggest provide `( )` for `?` operation

File: orchagent/crmorch.cpp [hunk 2]
```diff
@@ -839,3 +1115,10 @@ string CrmOrch::getCrmAclTableKey(sai_object_id_t id)
     ss << "ACL_TABLE_STATS:" << "0x" << std::hex << id;
     return ss.str();
 }
+
+string CrmOrch::getCrmDashAclGroupKey(sai_object_id_t id)
```
Comment (prsunny) [inline orchagent/crmorch.cpp:1119]: What is this function for? Can you please have a comment and an example for easy reference?

**Comment by prsunny** [comment]
Overall lgtm, minor comments

File: orchagent/dash/dashaclorch.cpp (hunk 1 above)
Comment (oleksandrivantsiv) [reply orchagent/dash/dashaclorch.cpp]: Done

File: orchagent/crmorch.cpp (hunk 2 above)
Comment (oleksandrivantsiv) [reply orchagent/crmorch.cpp:1119]: Added comment

**Comment by oleksandrivantsiv** [comment]
@prsunny the comments are resolved. Can you please review and merge the PR?

File: orchagent/crmorch.h [hunk 3]
```diff
@@ -63,6 +78,10 @@ class CrmOrch : public Orch
     void incCrmAclTableUsedCounter(CrmResourceType resource, sai_object_id_t tableId);
     // Decrement "used" counter for the per ACL table CRM resources (ACL entry/counter)
     void decCrmAclTableUsedCounter(CrmResourceType resource, sai_object_id_t tableId);
+    // Increment "used" counter for the per DASH ACL CRM resources (ACL group/rule)
+    void incCrmDashAclUsedCounter(CrmResourceType resource, sai_object_id_t groupId);
```
Comment (prabhataravind) [inline orchagent/crmorch.h:82]: Suggest using the same function naming convention for increment and decrement. Decrement has an additional Group within the name..

**Review by prabhataravind — COMMENTED** [review]
lgtm

File: orchagent/crmorch.h (hunk 3 above)
Comment (oleksandrivantsiv) [reply orchagent/crmorch.h:82]: Done

**Review by prabhataravind — APPROVED** [review]

**Pull request merged**